File: semgrep-rules/python/misc/testdata/sensitive_data_in_log.py
import logging

logger = logging.getLogger(__name__)


# ruleid: python-misc-sensitive-data-in-log
def log_password(password):
    logger.info("User password: %s", password)


# ruleid: python-misc-sensitive-data-in-log
def log_token(access_token):
    logging.debug("Token: %s", access_token)


# ruleid: python-misc-sensitive-data-in-log
def log_secret_fstring(secret_key):
    logger.error(f"Key = {secret_key}")


# ruleid: python-misc-sensitive-data-in-log
def log_api_key(api_key):
    logger.warn("API key: %s", api_key)


# ok: python-misc-sensitive-data-in-log
def log_normal(username):
    logger.info("User logged in: %s", username)


# ok: python-misc-sensitive-data-in-log
def log_message():
    logger.info("Please enter your password")


# ok: python-misc-sensitive-data-in-log
def log_user_id(user_id):
    logging.debug("Processing user: %s", user_id)

File: semgrep-rules/go/misc/testdata/sensitive_data_in_log.go
package testdata

import (
	"fmt"
	"log"
)

// ruleid: go-misc-sensitive-data-in-log
func logPassword(password string) {
	log.Printf("User password: %s", password)
}

// ruleid: go-misc-sensitive-data-in-log
func logToken(accessToken string) {
	log.Println("Token:", accessToken)
}

// ruleid: go-misc-sensitive-data-in-log
func logSecretKey(secretKey string) {
	fmt.Printf("Key: %s", secretKey)
}

// ruleid: go-misc-sensitive-data-in-log
func logApiKey(apiKey string) {
	log.Printf("API key = %s", apiKey)
}

// ok: go-misc-sensitive-data-in-log
func logNormal(username string) {
	log.Printf("User logged in: %s", username)
}

// ok: go-misc-sensitive-data-in-log
func logMessage() {
	log.Println("Please enter your password")
}

// ok: go-misc-sensitive-data-in-log
func logUserId(userId int) {
	log.Printf("Processing user: %d", userId)
}
